compiler/rustc_data_structures/src/sharded.rs

```diff
@@ -224,7 +223,7 @@ impl<K: Eq + Hash + Copy + IntoPointer> ShardedHashMap<K, ()> {
 
 #[inline]
 pub fn make_hash<K: Hash + ?Sized>(val: &K) -> u64 {
-    let mut state = FxHasher::default();
+    let mut state = ahash::AHasher::default();
```

Comment:
Ahash doesn't guarantee determinism of the hash between platforms and versions. This will likely lead to build artifacts that can't be reproduced unless rustc runs on the exact same architecture and likely lead to diagnostics changing between targets.